service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java

```diff
@@ -202,15 +191,13 @@ private ApiAzureLandingZoneResult fetchCreateAzureLandingZoneResult(String jobId
    * The initial assumption is that the cardinality of 1:1 between the cloud context and the LZ
    * is enforced in the create operation, therefore more than one LZ per azure context is not allowed.
    */
-  public String getLandingZoneId(AzureCloudContext azureCloudContext) {
-    return getFirstLandingZoneId(MapperUtils.LandingZoneTargetMapper.from(azureCloudContext))
+  public UUID getLandingZoneId(AzureCloudContext azureCloudContext) {
```

Comment:
See above comment -- we're currently querying for landing zones in the WSM `CloudContext` and taking the first one. 

However we probably want to refactor this to the desired model (I'm especially worried about missing permission checks involved in Azure cloud context creation).

Reply:
Yes, this call is upside down. Please add a TODO with the Jira ticket number that tracks when this gets done properly.